examples/platform/linux/AppMain.cpp

```diff
@@ -264,6 +264,26 @@ void EventHandler(const DeviceLayer::ChipDeviceEvent * event, intptr_t arg)
     {
         ChipLogProgress(DeviceLayer, "Receive kCHIPoBLEConnectionEstablished");
     }
+    else if ((event->Type == chip::DeviceLayer::DeviceEventType::kInternetConnectivityChange))
+    {
+        /* Network may not have been ready when mDNS attempted to broadcast an
+         * update, so restart the server here so it does it again,
+         * See https://github.com/project-chip/connectedhomeip/issues/27505
+         */
+        ChipLogProgress(DeviceLayer, "%s: DeviceEventType::kInternetConnectivityChange",
+            __func__);
+        chip::DeviceLayer::ChipDeviceEvent dns_restart_event;
+        dns_restart_event.Type = chip::DeviceLayer::DeviceEventType::kDnssdRestartNeeded;
+        CHIP_ERROR error = PlatformMgr().PostEvent(&dns_restart_event);
```

Comment:
Is there a reason this is not just restarting advertising directly?

Reply:
No, changed.